### DISCUSSION TITLE
Maintain page state between navigation (KeepAlive)

### DISCUSSION THREAD

**Discussion by ultravelocity** [lepoco/wpfui#617] [Q&A]
Is there any way to the maintain the state of the page when navigating between pages? I tried the KeepAlive property on the page but it didn't do anything. My app mainly has two sections that you can go back and forth between. I'd like to keep them both in memory and the retain the state of the each one.

**Comment by Mithicor** [comment]
I would like to know more about this as well. I see that there is a "NavigationCacheMode" property, which is what is used in WinUI to accomplish this. It doesn't seem to do anything for me currently, unless I am using it wrong?

In WinUI you would set that property on the page itself. Here it seems I can only set that property on the NavigationViewItem Control.

**Reply by Felix-CodingClimber** [reply]
I solved this by registering the viewModels as singleton
